rpc_util_test.go

```diff
@@ -290,3 +293,129 @@ func BenchmarkGZIPCompressor512KiB(b *testing.B) {
 func BenchmarkGZIPCompressor1MiB(b *testing.B) {
 	bmCompressor(b, 1024*1024, NewGZIPCompressor())
 }
+
+type testCompressor struct {
+	decompressedData []byte
+	errDecompress    error
+	customReader     io.Reader
+}
+
+func (m *testCompressor) Compress(w io.Writer) (io.WriteCloser, error) {
+	return nil, nil
+}
+
+func (m *testCompressor) Decompress(r io.Reader) (io.Reader, error) {
+	if m.errDecompress != nil {
+		return nil, m.errDecompress
+	}
+	if m.customReader != nil {
+		return m.customReader, nil
+	}
+	return bytes.NewReader(m.decompressedData), nil
+}
+
+func (m *testCompressor) Name() string {
+	return "testCompressor"
+}
+
+type ErrorReader struct{}
+
+func (e *ErrorReader) Read(p []byte) (n int, err error) {
+	return 0, errors.New("simulated io.Copy read error")
+}
+func compressData(data []byte) []byte {
+	var buf bytes.Buffer
+	// Use gzip as an example; replace with actual compressor if needed
+	gz := gzip.NewWriter(&buf)
+	_, _ = gz.Write(data)
+	_ = gz.Close()
+	return buf.Bytes()
+}
+func TestDecompress(t *testing.T) {
+	c := encoding.GetCompressor("gzip")
+	tests := []struct {
+		name                  string
+		compressor            encoding.Compressor
+		input                 mem.BufferSlice
+		maxReceiveMessageSize int
+		want                  mem.BufferSlice
+		compressedsize        int
+		error                 error
+	}{
+		{
+			name:       "Successful decompression",
```

Comment:
add more context to name. For example, in this case actual message size is less than allowed receive size. Keep it short though.